src/rpn2_geoclaw_vec.f90

```diff
@@ -0,0 +1,403 @@
+!       for the shallow water equations, accounting for wet dry boundary    !
+!       dry cells, wave speeds etc.                                         !
+!                                                                           !
+!           David George, Vancouver WA, Feb. 2009                           !
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+      use geoclaw_module, only: g => grav, drytol => dry_tolerance, rho
+      use geoclaw_module, only: earth_radius, deg2rad
+      use amr_module, only: mcapa
+
+      use storm_module, only: pressure_forcing, pressure_index
+      use statistics
+
+      implicit none
+
+      !input
+      integer maxm,meqn,maux,mwaves,mbc,mx,ixy
+
+      double precision  fwave(1-mbc:maxm+mbc, meqn, mwaves)
+      double precision  s(1-mbc:maxm+mbc, mwaves)
+      double precision  ql(1-mbc:maxm+mbc, meqn)
+      double precision  qr(1-mbc:maxm+mbc, meqn)
+      double precision  apdq(1-mbc:maxm+mbc,meqn)
+      double precision  amdq(1-mbc:maxm+mbc,meqn)
+      double precision  auxl(1-mbc:maxm+mbc,maux)
+      double precision  auxr(1-mbc:maxm+mbc,maux)
+
+      !local only
+      integer m,i,mw,maxiter,mu,nv
+      double precision wall(3)
+      !double precision fw(3,3)
+      !double precision sw(3)
+      double precision fw11, fw12, fw13, fw21, fw22, fw23, fw31, fw32, fw33
+      double precision sw1, sw2, sw3
+
+      double precision hR,hL,huR,huL,hvR,hvL,pL,pR
+      double precision bR,bL
+
+      double precision tw,dxdc
+      
+      double precision sqrt_ghL, sqrt_ghR
+
+      logical rare1,rare2
+      
+      interface
+        subroutine solve_rpn2(hL, hR, huL, huR, hvL, hvR, bL, bR, pL, pR,sw1,sw2,sw3,fw11,fw12,fw13,fw21,fw22,fw23,fw31,fw32,fw33) 
+            real(kind=8), intent(inout) :: sw1,sw2,sw3,fw11,fw12,fw13,fw21,fw22,fw23,fw31,fw32,fw33
+            real(kind=8), intent(inout) :: hL, hR, huL, huR, hvL, hvR, bL, bR, pL, pR
+        end subroutine
+      end interface
+
+      ! In case there is no pressure forcing
+      pL = 0.d0
+      pR = 0.d0
+      
+      !set normal direction
+      if (ixy.eq.1) then
+         mu=2
+         nv=3
+      else
+         mu=3
+         nv=2
+      endif
+
+      !loop through Riemann problems at each grid cell
+#if defined(_OPENMP)
+      !$OMP SIMD PRIVATE(hL,hR,huL,huR,hvL,hvR,bL,bR,pL,pR, &
+      !$OMP& fw11,fw12,fw13,fw21,fw22,fw23,fw31,fw32,fw33,sw1,sw2,sw3)
+#endif
+      do i=2-mbc,mx+mbc
+
+         !Riemann problem variables
+         hL = qr(i-1,1) 
+         hR = ql(i,1) 
+         huL = qr(i-1,mu) 
+         huR = ql(i,mu) 
+         bL = auxr(i-1,1)
+         bR = auxl(i,1)
+         pL = 0.d0
+         pR = 0.d0
+         if (pressure_forcing) then
+             pL = auxr(i-1,pressure_index)
+             pR = auxl(i, pressure_index)
+         end if
+
+         hvL=qr(i-1,nv) 
+         hvR=ql(i,nv)
+         
+         sw1 = 0.0d0
```

Comment:
You may want to add a comment as to why you needed to decompose this into scalars for future reference.

Reply:
Done!